src/java.base/share/classes/javax/crypto/MacSpi.java

```diff
@@ -101,6 +101,9 @@ protected abstract void engineInit(Key key,
      * process ByteBuffers more efficiently than byte arrays.
      *
      * @param input the ByteBuffer
+     *
+     * @exception NullPointerException if {@code input} is null
```

Comment:
nit: I recall @throws is preferred.

How about the update(...) methods in javax.crypto.Mac class? Perhaps the NPE there should be documented as well?

Reply:
@valeriepeng I totally agree about `@throws`. I was just going for consistency within the file. There are other `@exception` appearances.

Yes, I do agree about the `update(ByteBuffer)` in Mac as well; however this one throws `IllegalArgumentException` if the parameter is null. Or were you referring to another version of the `update` method?

Reply:
@valeriepeng see https://github.com/openjdk/jdk/pull/9859/commits/0bc0394e7a72f40b1f8f7cd4056b5906d4b89eda for the proposed change.

Let me know what you think about consistency within the files wrt `@throws` vs `@exception`.

Reply:
For this particular case, I would go for @throws. Moving toward the newer notation seems better.

Reply:
> @valeriepeng I totally agree about `@throws`. I was just going for consistency within the file. There are other `@exception` appearances.
> 
> Yes, I do agree about the `update(ByteBuffer)` in Mac as well; however this one throws `IllegalArgumentException` if the parameter is null. Or were you referring to another version of the `update` method?

Right, I should have not stated NPE, but rather the IAE caused by null argument. For completeness sake, perhaps we can cover NPE for Mac.update(byte[]) also? Since we are filing CSR already. It'd be great if this fix covers both Mac and MacSpi regarding null argument of the update(...)/engineUpdate(..) methods. Or, if that's a bit much, then we should strive to cover Mac.update(ByteBuffer)/MacSpi.engineUpdate(ByteBuffer) methods for completeness sake.